From 764f30b4c206790fb02d9144c0291615b93aa7f4 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:44:02 -0400
Subject: [ethernet] cxgb4: Fix for handling 1Gb/s SFP+ Transceiver Modules

Message-id: <1410475447-94392-104-git-send-email-svemuri@redhat.com>
Patchwork-id: 93879
O-Subject: [RHEL7.1 PATCH BZ 1124947 103/108] cxgb4: Fix for handling 1Gb/s SFP+ Transceiver Modules
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

We previously assumed that a Port's Capabilities and Advertised Capabilities
would never change from Port Initialization time.  This is no longer true
when we can have 10Gb/s and 1Gb/s SFP+ Transceiver Modules randomly swapped.

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 444018a7f1f46d588328498777cab5a9f4d29105)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 8ff46cf..b6e1304 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -3683,6 +3683,7 @@ int t4_handle_fw_rpl(struct adapter *adap, const __be64 *rpl)
    lc->link_ok = link_ok;
    lc->speed = speed;
    lc->fc = fc;
+   lc->supported = be16_to_cpu(p->u.info.pcap);
    t4_os_link_changed(adap, port, link_ok);
   }
   if (mod != pi->mod_type) {
-- 
1.7.1